var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -40,6 +41,10 @@ class Charliecloud(MakefilePackage):
     depends_on('py-sphinx',           type='build', when='+docs')
     depends_on('py-sphinx-rtd-theme', type='build', when='+docs')
 
+    # bash automated testing harness (bats)
+    variant('test', default=False, description='Install test suite dependencies')
+    depends_on('bats@0.4.0', type='run', when='+test')
```

Comment:
If you change this to:
```suggestion
    depends_on('bats@0.4.0', type='test')
```
then Spack will automatically install `bats` when `spack install --test` is used. No variant necessary.